### ISSUE TITLE
Confusion re: gradient flux and BCs

### ISSUE THREAD

**Issue by kmdeck** [CliMA/ClimateMachine.jl#1499]
# Description
I think it would be helpful to define what the gradient flux is in the how-to tutorial on how-to make a balance law. Since neumann BC on are on this variable, and there appears to be a correct choice, it would be good to spell it out more.

For example, in the soil water model, we have a diffusion-type eqn where the RHS is \nabla dot (K(Y) \nabla h(Y)). There was confusion initially about if nabla h(Y) could be the gradient flux (with BCs on this) vs if it should be K(Y) \nabla h(Y). 

also, some issue here:
https://clima.github.io/ClimateMachine.jl/latest/HowToGuides/Numerics/DGMethods/how_to_make_a_balance_law/#Reference-links

# For CLIMA Developers

- [ ] This feature can be added (if it cannot be, explain why in a comment below -- lack of technical expertise, not relevant to the scope of this project, too ambitious)
- [ ] There is a timeline for when this feature can be implemented
- [ ] The feature has been (or will be) implemented (Please link the PR)

**Comment by jiahe23** [comment]
Agreed! I suggest to create a high-level page for `balance law` (not to hide deeply inside in HowToGuide -> Numerics -> DGMethods). Inside the page, we could have a subpage `fundamental elements in a balance law` that contains the current introductions on varnames/methods, and another subpage `how-to-make-your-own-balance-law` that expands the current `General Remarks` page.
